Modules/posixmodule.c

```diff
@@ -3277,6 +3418,322 @@ os_lstat_impl(PyObject *module, path_t *path, int dir_fd)
 }
 
 
+#ifdef HAVE_STATX
+#define STATX_RESULT_CACHE_SLOTS 17
+typedef struct {
+    PyObject_HEAD
+    struct statx stx;
+#if STATX_RESULT_CACHE_SLOTS > 0
```

Comment:
isn't this always > 0? (the constant 17)

Reply:
It's defined this way to allow reviewers and future maintainers to more easily experiment with not having a cache (because member arrays can't be empty in C).  On reflection, it doesn't help much, because the no-cache configuration would change most of the getset defs to members for faster access, and after that, commenting out the member is trivial.